cmd/kubeadm/app/cmd/reset.go

```diff
@@ -63,19 +63,18 @@ func NewCmdReset(in io.Reader, out io.Writer) *cobra.Command {
 			ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(ignorePreflightErrors)
 			kubeadmutil.CheckErr(err)
 
-			if _, err := os.Stat(kubeConfigFile); !os.IsNotExist(err) {
-				client, err = getClientset(kubeConfigFile, false)
-				kubeadmutil.CheckErr(err)
+			var cfg *kubeadmapi.InitConfiguration
+			client, err = getClientset(kubeConfigFile, false)
+			if err == nil {
```

Comment:
This is removing the `os.Stat` along with the `kubeadmutil.CheckErr` which will fatal when called. This is a change in behavior that wasn't called out in the description. It seems like this might cause an NPE at some point later on in the code?

Also the error checking is a little funny here and could be reduced / an indentation level removed by flipping the logic around to look more go-ish:

```
client, err := ...
if err != nil {
  // log the thing
  // os.exit
}
cfg, err := ...
```

Reply:
@chuckha 

> depends on #74869
> please review only the second commit.

:)

> This is removing the os.Stat along with the kubeadmutil.CheckErr which will fatal when called. This is a change in behavior that wasn't called out in the description. It seems like this might cause an NPE at some point later on in the code?
> Also the error checking is a little funny here and could be reduced / an indentation level removed by flipping the logic around to look more go-ish:

we don't want to exit if `err != nil` and we cannot obtain a client but instead continue.
the is follow up logic has extra checks if `cfg != nil`. reason for that is that the lack of InitConfiguration is not fatal for reset.

Reply:
oops! my mistake.